src/nullablevector.jl

```diff
@@ -311,3 +311,14 @@ function Base.empty!(X::NullableVector)
     empty!(X.isnull)
     return X
 end
+
+notnullarray = Union{filter!(x -> !isa(x, Type{NullableArray}), subtypes(AbstractArray))...}
+notnullmatrix = typeintersect(notnullarray, AbstractMatrix)
+notnullvector = typeintersect(notnullarray, AbstractVector)
+
+Base.vcat{T <: notnullarray}(A::T, B::NullableArray) = Base.vcat(NullableArray(A), B)
```

Comment:
This is wasteful since it creates a copy. Maybe we can use this trick to force creating a `NullableArray`: `vcat(NullableArray{eltype(A)}(), A, B)`.